lib/gateway/context.js

```diff
@@ -1,7 +1,18 @@
 const vm = require('vm');
-
-function EgContextBase () {}
-
+const uuid62 = require('uuid-base62');
+function EgContextBase () {
+  this.traceId = uuid62.v4();
+}
+Object.defineProperty(EgContextBase.prototype, 'consumer', {
+  get () {
+    return this.req.user;
+  }
+});
+Object.defineProperty(EgContextBase.prototype, 'egContext', {
```

Comment:
oh, this is magic. 

`egContext.req.xx` should work same way as `req.xx`
this is why such reference exist

Important note: it should not be constructed on the fly, `vm` should receive same object whenever possible, to avoid building context